### PR TITLE
:art: :bricks: Improvements on delete block #241

### LINKED ISSUE #241: 🎨 🧱  Improvements on delete block
## User stories

Currently the delete bock works but we need a few enhancements to improve user experience.

---

Quick links: [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

---

### Improvement - More like a bug

Currently when a block is deleted, the connection to that block still exists on the database. So this can cause issues when the bot is trying to play through the blocks and hits a connection which does not lead to anywhere.

The connections linked to the block should also be deleted when the block is deleted

### Acceptance Criteria

1. Once I click on delete block icon, the block should dissapper from view. 
1. Once a block is deleted the connections to and from the block should me marked as deleted.

### PR DESCRIPTION
**Pull request by AnnaL001**
# Description
Improves the delete block functionality by marking its connections to and from, as deleted and removing them from the database.

Fixes #241 

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ import { _CreateOpenEndedQuestionBlockForm } from '../../model/open-ended-questi
 import { _CreateVideoInputBlockForm } from '../../model/video-input-block-form.model'
 
 import { BlockInjectorService } from '../../providers/block-injector.service';
+import { NewStoryService } from 'libs/features/convs-mgr/stories/home/src/lib/services/new-story.service';
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.ts:39]: Use domain imports: `@app/import-path`

File: libs/features/convs-mgr/stories/home/src/lib/services/new-story.service.ts [hunk 2]
```diff
@@ -115,6 +117,23 @@ export class NewStoryService implements OnDestroy {
...
+        // Mark the connections as deleted
+        connections.forEach(function(value){
+          value.deleted = true;
+        });
+        // Delete from store
+        this._blocksConnectionStore$$.removeMultiple(connections);
+      })
+    );
+  }
+
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/home/src/lib/services/new-story.service.ts:136]: It's best if this function is in the block connections store

File: libs/features/convs-mgr/stories/home/src/lib/services/new-story.service.ts [hunk 3]
```diff
@@ -115,6 +117,23 @@ export class NewStoryService implements OnDestroy {
...
+    const blockId = block.id?.toString() as string
+    let connections: StoryBlockConnection[] = []
+
+    this._sbS.add(
+      this._blocksConnectionStore$$.get().subscribe(_connections => {
+        connections = _connections.filter(conn => conn.sourceId.endsWith(blockId) || conn.targetId === blockId);
+        // Mark the connections as deleted
+        connections.forEach(function(value){
+          value.deleted = true;
+        });
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/home/src/lib/services/new-story.service.ts:130]: No need to mark the connection as deleted, if its going to be deleted from the store immediately after

**Review by ReaganCn — CHANGES_REQUESTED** [review]
Good work. Just some minor changes

File: libs/features/convs-mgr/stories/home/src/lib/services/new-story.service.ts (hunk 3 above)
Comment (AnnaL001) [reply libs/features/convs-mgr/stories/home/src/lib/services/new-story.service.ts]: Has been removed

File: libs/features/convs-mgr/stories/home/src/lib/services/new-story.service.ts (hunk 2 above)
Comment (AnnaL001) [reply libs/features/convs-mgr/stories/home/src/lib/services/new-story.service.ts]: Has been transferred to block connections store and block connections service

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.ts (hunk 1 above)
Comment (AnnaL001) [reply libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.ts]: The import has been removed

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.ts [hunk 4]
```diff
@@ -36,6 +36,7 @@ import { _CreateOpenEndedQuestionBlockForm } from '../../model/open-ended-questi
 import { _CreateVideoInputBlockForm } from '../../model/video-input-block-form.model'
 
 import { BlockInjectorService } from '../../providers/block-injector.service';
+import { BlockConnectionsService } from '@app/state/convs-mgr/stories/block-connections';
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.ts:39]: Domain imports should come before local @app before '../../imports'

**Review by ReaganCn — CHANGES_REQUESTED** [review]
Good job, just one import fix

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.ts (hunk 4 above)
Comment (AnnaL001) [reply libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.ts]: Done

**Review by ReaganCn — APPROVED** [review]

**Pull request merged**